pkg/operator/controller/ingress/controller.go

```diff
@@ -617,6 +622,27 @@ func setDefaultPublishingStrategy(ic *operatorv1.IngressController, platformStat
 					statusLB.ProviderParameters.IBM.Protocol = specProtocol
 					changed = true
 				}
+			case operatorv1.OpenStackLoadBalancerProvider:
+				// The only provider parameter that is supported
+				// for OpenStack is the FloatingIP parameter.
+				var statusFloatingIP string
```

Comment:
So I hate to bring this up, but I have been pushing for new APIs to have their status counterpart reflect the *actual* state. You have followed the traditional path of `status` representing the desired state of the IngressController (user desire from`spec` + defaulting).

For `Subnets`, `EIPAllocations`, `AllowedSourceRanges`, we have started to pave this new path for status to reflect actual state. But I have mixed feelings about whether we did this prematurely because:
1. `status.endpointPublishingStrategy...` has mixed purpose status (some actual, some desired). That's probably bad and confusing UX (we probably should have forked into new status structure....)
2. The Ingress Operator spec & status related code is splitting into two logic "code flows" (complexity is :chart_with_upwards_trend:)

And we need to consider the priority of your RFE here. But when we start using the new `FloatingIP` status field, the API Team has said a change in "behavior" of status is incompatible (i.e. we are locked in). 

I'm not against maintaining the traditional `status` patterns here given the circumstances. Rewriting this code would be quite painful and time consuming.  But I'd like to have consensus from our team on what our plan is. I'm thinking we should continue having `status.endpointPublishingStrategy` traditionally as *desired* as you've done here, but we should have a backlog task to add a new `status` struct for representing *actual* values to keep things organized. @Miciah thoughts?

Reply:
I'll let you decide how to want to deal with that. I'm happy either way, I can leave it like this or patch `IngressStatusesEqual` to add this new field. And I can also create that task and I could be assigned to it for the next release if you want.

Reply:
@EmilienM I had a long conversation with @Miciah about IngressController status today, with this PR in mind. We arrived at the conclusion that we should continue the pattern for status set forth in syncIngressControllerStatus: https://github.com/openshift/cluster-ingress-operator/blob/7fef44643ba04946a23b591a312cb46dbc25aac5/pkg/operator/controller/ingress/status.go#L77-L82
 
By this, I mean status should reflect the *actual* state, not the desired state. Mind trying this out? This mean a couple of things will change in your PR:
1. You shouldn't be making any changes to `setDefaultPublishingStrategy` or `setDefaultProviderParameters` (i.e. your current changes should be remove)
2. You'll need to add some logic in `syncIngressControllerStatus`. I recommend following the existing patterns set forth by Subnets & EIPAllocations.
3. You'll need to update any references that consume status as the desired state, and ensure you consume status as the actual state (e.g. your have & want logic in `loadBalancerServiceIsProgressing`)
4. Don't forget about updating `IngressStatusesEqual` too
5. I'd like to put a note in the docs of `setDefaultPublishingStrategy` about this pattern we are changing, that way we can communicate our desired pattern to future developers.

I might be forgetting something, but hope this helps with the transition. Sorry for churn.

Reply:
1. :heavy_check_mark: 
2. :heavy_check_mark: 
3. :heavy_check_mark: but no tests yet, I want to make sure I'm on the right path first.
4. :heavy_check_mark: but no tests yet, I want to make sure I'm on the right path first.
5. I'll need your help for this one or maybe we can do it via its own PR, please.